File: Templates/Project Templates/macOS/Application Extension/Persistent Token Extension.xctemplate/TokenDriver.m
//___FILEHEADER___

#import "Token.h"

@implementation TokenDriver

- (TKToken *)tokenDriver:(TKTokenDriver *)driver tokenForConfiguration:(TKTokenConfiguration *)configuration error:(NSError **)error {
    return [[Token alloc] initWithTokenDriver:self instanceID:configuration.instanceID];
}

@end

File: Templates/Project Templates/macOS/Application Extension/Persistent Token Extension.xctemplate/TokenDriver.swift
//___FILEHEADER___

import CryptoTokenKit

class TokenDriver: TKTokenDriver, TKTokenDriverDelegate {

    func tokenDriver(_ driver: TKTokenDriver, tokenFor configuration: TKToken.Configuration) throws -> TKToken {
        return Token(tokenDriver: self, instanceID: configuration.instanceID)
    }

}
